static/js/components/search/SemesterRefinementOption_test.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+/* global SETTINGS: false */
+import React from "react"
+import { assert } from "chai"
+import _ from "lodash"
+import sinon from "sinon"
+import ReactTestUtils from "react-dom/test-utils"
+
+import SemesterRefinementOption from "./SemesterRefinementOption"
+import { makeStrippedHtml } from "../../util/util"
+
+describe("SemesterRefinementOption", () => {
+  const renderSemesterOption = props =>
+    makeStrippedHtml(<SemesterRefinementOption {...props} />)
+
+  const onClick = sinon.stub()
+  const props = {
+    label:   "2016 - Summer",
+    active:  false,
+    onClick: onClick,
+    count:   42
+  }
+
+  const renderFullSemesterOption = props =>
+    ReactTestUtils.renderIntoDocument(<SemesterRefinementOption {...props} />)
```

Comment:
could you use enzyme here instead of `ReactTestUtils`?

Reply:
we are mocking a property here, ReactTestUtils is used in all these cases in other search apps.

Reply:
what property are you mocking? is there a reason you can't write the test with enzyme? we used to use ReactTestUtils for tests but some time ago we switched to writing new tests with enzyme, so if possible we should use that. I could be missing something, but reading through this file I don't see anything that couldn't be testing with enzyme.

Reply:
it is onClick method which is prop of SemesterOption

Reply:
https://github.com/mitodl/micromasters/pull/3936/files#diff-7169af83398278868d9e402c51a0f669R14

Reply:
yeah what you're doing here is fully supported by using enzyme, which uses ReactTestUtils under the hood. As I said, we used to use ReactTestUtils directly for writing tests, but this is deprecated and we don't use it for writing new tests. For new tests we add we want to just use enzyme. There are lots of tests in MM written with ReactTestUtils which were written before we switched, including a lot of the tests for the learner search components.

Here's how you could rewrite these tests to use enzyme, as a suggestion:

```js
// @flow
/* global SETTINGS: false */
import React from "react"
import { assert } from "chai"
import sinon from "sinon"
import { shallow } from "enzyme"

import SemesterRefinementOption from "./SemesterRefinementOption"

describe("SemesterRefinementOption", () => {
  const renderSemesterOption = () =>
    shallow(<SemesterRefinementOption {...props} />)

  let props

  beforeEach(() => {
    props = {
      onClick: sinon.stub(),
      label:   "2016 - Summer",
      active:  false,
      count:   42
    }
  })

  it("should render a label", () => {
    const option = renderSemesterOption()
    assert.include(option.text(), "2016 - Summer")
  })

  it("should render a semester placeholder, given no semester code", () => {
    props.label = ""
    const option = renderSemesterOption()
    assert.include(option.text(), "N/A")
  })

  it("should display the result count for the option", () => {
    const option = renderSemesterOption()
    assert.include(option.text(), "42")
  })

  it("should bind an onClick handler", () => {
    renderSemesterOption().simulate("click")
    assert(props.onClick.called, "onClick handler wasn't called")
  })
})
```

Reply:
Done